File: WindowsFormsApplication1/WindowsFormsApplication1/Examples/Sinus.cpp
#include "TXLib.h"

struct Person
{
    double x;
    double y;
    double angle;
    double nomer_kadra;
    HDC texture;
    int MID_x;
    int MID_y;
    int rad;
    int skorost;
    int gr_dvigx;
    int gr_dvigy;
    int nach_dv;
    int ampl_y;
    int ampl_x;
    double BeginTime;
    double EndTime;
    double Time;
};

Person per;
Person per1;


int main()
{
    txCreateWindow(800, 600);

    per.texture = txLoadImage("Pictures\\Personaj.bmp");

    per.x = 50;
    per.y = 50;
    per.nomer_kadra = 0;
    per.nach_dv = 300;
    per.ampl_y = 150;
    per.ampl_x = 10;


    per1.x = 50;
    per1.y = 50;
    per1.angle = 0;
    per1.nomer_kadra = 0;
    per1.texture = txLoadImage("Pictures\\Personaj.bmp");
    per1.MID_x = 500;
    per1.MID_y = 300;
    per1.rad = 200;
    per1.skorost = 10;


    while (!GetAsyncKeyState(VK_ESCAPE))
    {
        txSetColor(TX_RED);
        txSetFillColor(TX_RED);
        txRectangle(0, 0, txGetExtentX(), txGetExtentY());

        per.x++;
        per.y = per.nach_dv + per.ampl_y * sin(per.x / per.ampl_x);

        txTransparentBlt(txDC(), per.x, per.y, 55, 86, per.texture, 55 * per.nomer_kadra, 0, RGB(0, 255, 255));

        per.nomer_kadra++;

        if (per.nomer_kadra > 2)
        {
            per.nomer_kadra = 0;
        }


        per1.angle++;
        per1.x = per1.MID_x + per1.rad * cos (per1.angle / per1.skorost);
        per1.y = per1.MID_y + per1.rad * sin (per1.angle / per1.skorost);

        txTransparentBlt(txDC(), per1.x, per1.y, 55, 86, per1.texture, 0, 0, RGB(0, 255, 255));


        txSleep(10);
        txSleep(10);
    }

    txDeleteDC(per.texture);
    txDeleteDC(per1.texture);

    return 0;
}

File: WindowsFormsApplication1/WindowsFormsApplication1/Sinus.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;

namespace WindowsFormsApplication1
{
    public class Sinus
    {
        public static void CreatePerson(string filename, string name, string xBeg)
        {
            File.AppendAllText(filename, "    Person " + name + ";" + Environment.NewLine);
            File.AppendAllText(filename,                              Environment.NewLine);
            File.AppendAllText(filename, "    " + name + ".texture = txLoadImage(\"Pictures\\\\Personaj.bmp\");" + Environment.NewLine);
            File.AppendAllText(filename, "    " + name + ".x = " + xBeg + ";" + Environment.NewLine);
            File.AppendAllText(filename, "    " + name + ".y = " + xBeg + ";" + Environment.NewLine);
            File.AppendAllText(filename, "    " + name + ".nomer_kadra = 0;" + Environment.NewLine);
            File.AppendAllText(filename, "    " + name + ".nach_dv = " + name + ".y;" + Environment.NewLine);
            File.AppendAllText(filename, "    " + name + ".ampl_y = 150;" +    Environment.NewLine);
            File.AppendAllText(filename, "    " + name + ".ampl_x = 10;" +     Environment.NewLine); 
        }

        public static void MovePerson(string filename, string name)
        {
            File.AppendAllText(filename, "        " + name + ".x++;" + Environment.NewLine);
            File.AppendAllText(filename, "        " + name + ".y = " + name + ".nach_dv + " + name + ".ampl_y * sin(" + name + ".x / " + name + ".ampl_x);" + Environment.NewLine);
            File.AppendAllText(filename, "        txTransparentBlt(txDC(), " + name + ".x, " + name + ".y, 55, 86, " + name + ".texture, 55 * " + name + ".nomer_kadra, 0, RGB(0, 255, 255));" + Environment.NewLine);
            File.AppendAllText(filename, Environment.NewLine);
            File.AppendAllText(filename, "        " + name + ".nomer_kadra++;" + Environment.NewLine);
            File.AppendAllText(filename, "        if (" + name + ".nomer_kadra > 2)" + Environment.NewLine);
            File.AppendAllText(filename, "        {" + Environment.NewLine);
            File.AppendAllText(filename, "            " + name + ".nomer_kadra = 0;" + Environment.NewLine);
            File.AppendAllText(filename, "        }" + Environment.NewLine);
        }
    }
}
